packages/babel-generator/src/generators/methods.ts

```diff
@@ -151,12 +137,18 @@ export function ArrowFunctionExpression(
   this.print(node.body, node);
 }
 
-function hasTypes(node, param) {
-  return (
+function hasTypesOrComments(
+  node: t.ArrowFunctionExpression,
+  param: t.Identifier,
+): boolean {
+  return !!(
     node.typeParameters ||
     node.returnType ||
+    // @ts-expect-error
+    node.predicate ||
```

Comment:
This doesn't type-check because `predicate` isn't declared on `ArrowFunctionExpression`.